proxy/coco/src/peer.rs

```diff
@@ -715,6 +752,96 @@ mod test {
         Ok(())
     }
 
+    /// Verify that asking the network for an unkown urn returns no providers.
+    #[tokio::test]
+    async fn get_urn_providers_none() -> Result<(), Error> {
+        let tmp_dir = tempfile::tempdir().expect("failed to create temdir");
+        let key = SecretKey::new();
+        let config = config::default(key, tmp_dir.path())?;
+        let api = Api::new(config).await?;
+        let unkown_urn = uri::RadUrn {
+            id: Hash::hash(b"project0"),
+            proto: uri::Protocol::Git,
+            path: "user/imperative-language"
+                .parse::<uri::Path>()
+                .map_err(Error::from)?,
+        };
+
+        let res = timeout(
+            Duration::from_secs(5),
+            api.providers(unkown_urn).await.await.next(),
```

Comment:
Why is there a double await here?

Reply:
Having https://github.com/radicle-dev/radicle-upstream/pull/875#discussion_r485619418 in mind, the double await comes from first unboxing the future and then operate on it. Happy to reconsider that approach :hand: